test/JDBC/replication/datatype_test.mix

```diff
@@ -0,0 +1,159 @@
+-- Create tables on both publisher and subscriber
+-- with all the supported datatypes.
+
+-- tsql
+CREATE TABLE datatype_test_t1 (
+    c1  BIGINT  NOT NULL PRIMARY KEY
+    , c2    BINARY(20) NOT NULL
+    , c3    BIT NOT NULL
+    , c4    CHAR(20)   NOT NULL
+    , c5    DATE    NOT NULL
+    , c6    DATETIME    NOT NULL
+    , c7    DATETIME2   NOT NULL
+    , c8    DATETIMEOFFSET  NOT NULL
+    , c9    DECIMAL(8,4)    NOT NULL
+    , c10   FLOAT   NOT NULL
+    , c11   IMAGE   NOT NULL
+    , c12   INT NOT NULL
+    , c13   MONEY   NOT NULL
+    , c14   NCHAR(20)  NOT NULL
+    , c15   NTEXT   NOT NULL
+    , c16   NUMERIC(8,4)    NOT NULL
+    , c17   NVARCHAR(20)   NOT NULL
+    , c18   REAL    NOT NULL
+    , c19   SMALLDATETIME   NOT NULL
+    , c20   SMALLINT    NOT NULL
+    , c21   SMALLMONEY  NOT NULL
+    , c22   SQL_VARIANT  NOT NULL
+    , c23   SYSNAME NOT NULL
+    , c24   TEXT    NOT NULL
+    , c25   TIME    NOT NULL
+    , c26   TINYINT NOT NULL
+    , c27   UNIQUEIDENTIFIER    NOT NULL
+    , c28   VARBINARY(20)  NOT NULL
+    , c29   VARCHAR(20)    NOT NULL
+    , c30   XML NOT NULL)
+GO
+
+-- tsql port=8199
+CREATE TABLE datatype_test_t1 (
```

Comment:
Can we add negative tests where schema of target is a bit different.
1. Implicit casting
2. Throws error during insert maybe due to constraint issues. Whats the behaviour in that case?